codespell_lib/_codespell.py

```diff
@@ -429,10 +431,7 @@ def build_dict(filename, misspellings, ignore_words):
     with codecs.open(filename, mode='r', encoding='utf-8') as f:
         for line in f:
             [key, data] = line.split('->')
-            # TODO for now, convert both to lower. Someday we can maybe add
-            # support for fixing caps.
             key = key.lower()
-            data = data.lower()
```

Comment:
Unfortunately this is overly simplistic. Consider:
`m_host_ip = ip`

With this change this will become:
`m_host_IP = IP`

Which we probably don't want!

Technically currently a constant called say `GITTHUB` might currently get corrected to `github`, which isn't ideal but at least we've fixed the constant.

I'm not sure there's actually an easy fix for this in non-interactive mode, apart from just flagging it as a potential capitalisation issue. Obviously we can offer it in interactive mode.

Reply:
Wait? This project also does spellchecking in source code, i.e., function names and variables? I was operating under the assumption that it's mainly aimed at comments.

Reply:
Yeah hence the name @cristiklein . :smile: 

Broadly by default it's not clever enough to differentiate, although it may be possible to hook it into existing parsers to handle that sort of thing.

For example if you search here for sic you'll see most of the external facing/API changes that have been caught (primarily) by codespell (it's also caught a load of internal ones which aren't mentioned):
https://github.com/OpenLightingProject/ola/blob/master/NEWS